src/sentry/integrations/slack/integration.py

```diff
@@ -157,3 +165,164 @@ def build_integration(self, state):
+
+            return render_to_response(
+                template="sentry/integrations/slack-reauth-introduction.html",
+                context={
+                    "next_url": "%s%s" % (absolute_uri("/extensions/slack/setup/"), next_param),
+                },
+                request=request,
+            )
+
+        if "show_verification_results" in request.GET:
+            return pipeline.next_step()
+
+        # if we dont have the integration_id we dont care about the
+        # migration path, skip straight to install
+        pipeline.state.step_index = 2
+        return pipeline.current_step()
+
+
+class SlackReAuthChannels(PipelineView):
+    """
+        This pipeline step handles making requests to Slack and
+        displaying the channels (if any) that are problematic:
+
+        1. private
+        2. removed
+        3. unauthorized
+
+        Any private channels in alert rules will also be binded
+        to the pipeline state to be used later.
+
+    """
+
+    def dispatch(self, request, pipeline):
+        if "start_migration" in request.GET:
+            return pipeline.next_step()
+
+        next_url_param = "?start_migration"
+
+        try:
+            channels = _request_channel_info(pipeline)
+        except IntegrationError as error:
+            return pipeline.error(error)
+
+        return render_to_response(
+            template="sentry/integrations/slack-reauth-details.html",
+            context={
+                "next_url": "%s%s" % (absolute_uri("/extensions/slack/setup/"), next_url_param),
+                "private": channels["private"],
+                "no_permission": channels["no_permission"],
+                "not_found": channels["channel_not_found"],
+            },
+            request=request,
+        )
+
+
+def _request_channel_info(pipeline):
+    channels = pipeline.fetch_state("all_channels")
+    integration_id = pipeline.fetch_state("integration_id")
+
+    try:
+        integration = Integration.objects.get(id=integration_id, provider="slack",)
+    except Integration.DoesNotExist:
+        raise IntegrationError("Could not find Slack integration.")
+
+    channel_responses = defaultdict(lambda: set())
+    for channel in channels:
+        payload = {
+            "token": integration.metadata["access_token"],
+            "channel": channel["id"],
+        }
+        client = SlackClient()
+        try:
+            resp = client.post("/conversations.info", data=payload)
+            # TODO(meredith): subclass the ApiError and make a SlackApiError so we can
+            # reraise the other errors
+        except ApiError as e:
+            # adds the channel to our dict grouped by the error message which could
+            # be any of the following found under the 'errors' section found in
+            # https://api.slack.com/methods/conversations.list
+            channel_responses[e].add(Channel(channel["name"], channel["id"]))
```

Comment:
@MeredithAnya Is it possible for there to be any other types of error other than `no_permission` or `channel_not_found`? We will end up ignoring anything else. No need to fix it in this PR since you have a follow up one coming.